Add newest release notes (#179)
elastisys/compliantkubernetes: docs/release-notes.md

# Release Notes

## Compliant Kubernetes
<!-- BEGIN TOC -->
- [v0.17.0](#v0170) - 2021-06-29
- [v0.16.0](#v0160) - 2021-05-27
- [v0.15.0](#v0150) - 2021-05-05
<!-- END TOC -->

!!!note
    For more details check out the full [changelog](https://github.com/elastisys/compliantkubernetes-apps/blob/main/CHANGELOG.md) and the [upgrade guide](https://github.com/elastisys/compliantkubernetes-apps/tree/main/migration) for instructions needed to upgrade between versions.

### v0.17.0

Released 2021-06-29.

Changes:

> - Dex updated from its deprecated chart to the new upstream.

Reviewer: My understanding is that the user-facing change log should only capture changes relevant for the user, especially required actions. Does updating Dex change anything in the user experience? Does this change require any user action? If not, this line should be removed. If yes, then this line should be augmented with what has actually changed and what actions the user needs to take.

Made up example:

```suggestion
- The single-sign-on UI provided by Dex looks slightly differently due to an upgrade. No user action is required."
```

Author: Yeah, it might have been because it was a bit loosely defined that I've missed the mark. :sweat_smile:
I mainly set out to simplify, condense and make it more digestible, but it makes more sense to limit the scope to keep it relevant for the user. 

Should it only capture the user as defined under the user guide overview, or should it also capture a bit from the perspective of CISO and administrators? (Like when it comes to additions to Dex; group support, static user, etc.)

Reviewer: I believe that our administrators' needs are already captured by the "full" release notes.

Please focus only on the user as defined under the [user guide overview](https://compliantkubernetes.io/user-guide/) for now. Notice that we recently clarified their role a bit.

If it helps, keep in mind that we might want to send the changelog to our customers either *before* or *after* an Compliant Kubernetes upgrade. (This is currently to be decided.) Hence, the focus on user impact and actions required.

Author: I've written a new version from the perspective of the user. Hopefully it is a bit more in line with what you imagined.

> - NFS storage provisioner updated from its deprecated to the new upstream.

Reviewer: This doesn't feel user facing.

Author: It has been removed, mostly. Dex is changed similar to the comment above.

> - Monitoring stack updated with new versions of Prometheus and Grafana to resolve issues with some dashboards.

Reviewer: I would prefer:

```suggestion
- Some issues with some dashboards (which?) have been resolved. No user action is required.
```

> - Cert-manager updated.

Reviewer: Has the API changed? Which cert-manager APIs are deprecated as a result of this? Which APIs are available?

(I guess not, since we only really require the user to make some Ingress annotations.)

# Release Notes

## Compliant Kubernetes
<!-- BEGIN TOC -->
- [v0.17.0](#v0170) - 2021-06-29
- [v0.16.0](#v0160) - 2021-05-27
- [v0.15.0](#v0150) - 2021-05-05
<!-- END TOC -->

!!!note
    For a more detailed look check out the full [changelog](https://github.com/elastisys/compliantkubernetes-apps/blob/main/CHANGELOG.md).

### v0.17.0

Released 2021-06-29.

Changes:

> - The analythics and visualisation tool Grafana has been updated to a new major version of 8.x.x. This introduces new features and fixes, as well as some possibly breaking changes. See their [release notes](https://grafana.com/docs/grafana/v8.0/whatsnew/whats-new-in-v8-0/) for more information.

Reviewer: ```suggestion
- The dashboard tool Grafana has been updated to a new major version of 8.x.x. This introduces new features and fixes, as well as some possibly breaking changes. See their [release notes](https://grafana.com/docs/grafana/v8.0/whatsnew/whats-new-in-v8-0/) for more information.
```

Author: Changed.
